core/templates/dev/head/pages/exploration_player/AudioControlsDirective.js

```diff
@@ -0,0 +1,120 @@
+      },
+      templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+        '/pages/exploration_player/' +
+        'audio_controls_directive.html'),
+      controller: [
+        '$scope', 'AudioTranslationManagerService', 'AudioPlayerService',
+        function(
+            $scope, AudioTranslationManagerService, AudioPlayerService) {
+          var filenameOfLastStartedAudio;
+
+          $scope.AUDIO_SETTINGS_BUTTON_IMAGE_URL = (
+            UrlInterpolationService.getStaticImageUrl(
+              '/icons/settings.svg'));
+
+          var PLAY_AUDIO_BUTTON_IMAGE_URL = (
+            UrlInterpolationService.getStaticImageUrl(
+              '/icons/speaker-not-playing.svg'));
+
+          var PAUSE_AUDIO_BUTTON_IMAGE_URL = (
+            UrlInterpolationService.getStaticImageUrl(
+              '/icons/speaker-playing.svg'));
+
+          $scope.REWIND_AUDIO_BUTTON_IMAGE_URL = (
+            UrlInterpolationService.getStaticImageUrl(
+              '/icons/rewind-five.svg'));
+
+          $scope.playPauseAudioTranslation = function() {
+            // TODO(tjiang11): Change from on-demand loading to pre-loading.
+
+            // TODO(tjiang11): On first play, ask learner to pick language
+            // and subsequently for confirmation to use bandwidth 
+            // to download audio files.
+
+            $scope.extraAudioControlsAreShown = true;
+
+            if (!AudioPlayerService.isPlaying()) {
+              if (AudioPlayerService.isTrackLoaded() &&
+                  isRequestForSameAudioAsLastTime()) {
+                AudioPlayerService.play();
+              } else {
+                loadAndPlayAudioTranslation();
+              }
+            } else {
+              AudioPlayerService.pause();
+              if (!isRequestForSameAudioAsLastTime()) {
+                loadAndPlayAudioTranslation();
+              }
+            }
+          };
+
+          var isRequestForSameAudioAsLastTime = function() {
+            var requestedAudioTranslation =
+              $scope.audioTranslations[
+                AudioTranslationManagerService.getCurrentAudioLanguageCode()];
+            return filenameOfLastStartedAudio ==
+              requestedAudioTranslation.filename;
+          };
+
+          var loadAndPlayAudioTranslation = function() {
+            var currentAudioLanguageCode =
+              AudioTranslationManagerService.getCurrentAudioLanguageCode();
+
+            // TODO(tjiang11): If audio translation is not available
+            // in the current language, then inform the learner with
+            // a piece of text below the audio controls.
+            var audioTranslation =
+              $scope.audioTranslations[currentAudioLanguageCode];
+
+            if (audioTranslation) {
+              filenameOfLastStartedAudio = audioTranslation.filename;
+              AudioPlayerService.load(
+                audioTranslation.filename).then(function() {
+                  AudioPlayerService.play();
```

Comment:
Incidentally, what happens if this is called while some other audio is already playing?